### ISSUE TITLE
System.Text.Json, addition needed for JsonWriterOptions, to avoid URI serialization issue

### ISSUE THREAD

**Issue by Thorium** [dotnet/runtime#95182]
If you use System.Text.Json with JsonWriterOptions, by default settings (Indented off), it will create a following JSON:
{"url":"https://google.com","prop":"x"}

...which looks correct to me, but as you can see, the major automatic parsers like GitHub, Google, Microsoft, etc fail to end the url link to the end-quote, and treat the URL as combining all the rest of the stuff going as part of that url. And as you can see, the full url is not working.

If a single space is added after the comma, then behold, the parsers are clever now:
{"url":"https://google.com", "prop":"x"}

If you do Indented=true, then there is way too much unnecessary characters for a production system.
Could we consider adding a serialization option of "a space after comma", please?

**Comment by gregsdennis** [comment]
This, to me, seems like a bug in their parsers.  STJ is doing it right.

**Comment by Thorium** [comment]
if all the parsers are doing it wrong, we'd need to adopt...

**Comment by gregsdennis** [comment]
I disagree.  Deviation from standard should never become common practice.

**Comment by eiriktsarpalis** [comment]
Could you share an example of a JSON parser that is doing this? It seems pretty flagrant.

**Comment by vcsjones** [comment]
> major automatic parsers like GitHub

GitHub is not “parsing” the JSON in your example. It’s just a markdown rendering that automatically converts links in to clickable links. Our markdown pipeline does not look in issue content for “does this look like JSON?” and try to interpret it.

I would expect someone to use a code fence in this situation to avoid this.

```json
{"url":"https://google.com","prop":"x"}
```

**Comment by Thorium** [comment]
Adding one spacing option that is still within the standard syntax shouldn't be too hard...

I don't even mind Github but the same is done by Word, Outlook and others.

My problem is that I provide WebApi returning JSON and the customers of that API complain the URLs being wrong, and when things are escalated to manager level, the managers clearly see the URLs being wrong. I would expect the case not to be unique.

**Comment by eiriktsarpalis** [comment]
> Adding one spacing option that is still within the standard syntax shouldn't be too hard...

It's adding yet another source of branching in the hot path though. We might consider it if this were some kind of web standard, but it seems to me that the only motivating use case is to provide a workaround for a specific bug in tools that weren't designed to display JSON in the first place. And per @vcsjones's comment it's not even solving the problem in its entirety.

**Comment by Thorium** [comment]
> Does Json.NET have a formatting option that works around the issue?

If you do `Newtonsoft.Json.Formatting.None`, it's as compact and fast as it can, 
but if you do `Newtonsoft.Json.Formatting.Indented`, you can control everything:
https://www.newtonsoft.com/json/help/html/Properties_T_Newtonsoft_Json_JsonTextWriter.htm
